File: ai-complaint-model/test_predict.py
import requests
import json

url = "http://localhost:5001/predict"
data = {"text": "There is a huge pothole on the main road causing traffic"}
headers = {"Content-Type": "application/json"}

try:
    response = requests.post(url, json=data, headers=headers)
    print("Status Code:", response.status_code)
    try:
        json_resp = response.json()
        print("Keys:", list(json_resp.keys()))
        print("Confidence:", json_resp.get("confidence"))
    except Exception as e:
        print("JSON Error:", e)
        print("Raw:", response.text)
except Exception as e:
    print("Error:", e)

File: civic-sense-backend/test_predict.js
const axios = require('axios');
require('dotenv').config();

async function test() {
    const aiUrl = process.env.TEXT_AI_URL || 'http://localhost:5001';
    console.log(`Using AI URL: ${aiUrl}`);

    try {
        console.log("Testing Pothole...");
        const res1 = await axios.post(`${aiUrl}/predict`, { text: "Huge pothole on the main road" });
        console.log("Pothole:", res1.data);

        console.log("Testing Garbage...");
        const res2 = await axios.post(`${aiUrl}/predict`, { text: "Garbage has not been collected for days" });
        console.log("Garbage:", res2.data);

        console.log("Testing Streetlight...");
        const res3 = await axios.post(`${aiUrl}/predict`, { text: "Streetlight is broken" });
        console.log("Streetlight:", res3.data);

    } catch (e) {
        console.error("Error:", e.message);
        if (e.response) console.error("Data:", e.response.data);
    }
}

test();
